src/os/bluestore/BlueStore.cc

```diff
@@ -4454,6 +4445,23 @@ int BlueStore::_open_bdev(bool create)
   return r;
 }
 
+void BlueStore::_validate_bdev()
+{
+  assert(bdev);
+  assert(min_alloc_size); // _get_odisk_reserved depends on that
+  uint64_t dev_size = bdev->get_size();
+  if (dev_size < 
+    _get_ondisk_reserved() + cct->_conf->bluestore_bluefs_min) {
+    dout(1) << __func__ << " main device size " << byte_u_t(dev_size)
+            << " is too small, disable bluestore_bluefs_min for now"
```

Comment:
I don't think that a tiny device should be allowed, and it would be surprising for the OSD to be disabling `bluestore_bluefs_min` on the fly like this.

Would be OK if we just let it fail, explain that the problem is that the device is not big enough, and that changing `bluestore_bluefs_min` would help if wanting to forcefully use a small device

Reply:
@alfredodeza - most probably you're right and this makes no sense but it's been there originally. I just took superblock size into account and refactored code a bit...
@liewegas - do you think we still need this logic resetting bluestore_bluefs_min parameter?

Reply:
I think it's fine to keep.  bluestore_bluefs_min is about how much of a large device to keep in bluefs's hands.  I think it makes sense to adapt when trying to create a tiny volume.  OTOH this is only going to come up in testing situations, so if we remove this then there are probably a bunch of test failures to mop up as a result, but nobody in the real world should actually notice.

Reply:
I don't think there are test failures today though (they wouldn't work as it is now), this came from someone in the community trying to actually use 1gb for a data bluestore OSD. I would prefer an explicit failure and proper test configuration rather than automatic reconfiguration.